core/deployment/src/main/java/io/quarkus/deployment/pkg/steps/DockerBuildStep.java

```diff
@@ -0,0 +1,53 @@
+
+package io.quarkus.deployment.pkg.steps;
+
+import java.util.Optional;
+
+import org.jboss.logging.Logger;
+
+import io.quarkus.deployment.annotations.BuildStep;
+import io.quarkus.deployment.builditem.ApplicationInfoBuildItem;
+import io.quarkus.deployment.pkg.builditem.DockerBuild;
+import io.quarkus.deployment.pkg.builditem.DockerImageBuildItem;
+import io.quarkus.deployment.pkg.builditem.DockerImageResultBuildItem;
+import io.quarkus.deployment.pkg.builditem.JarBuildItem;
+import io.quarkus.deployment.pkg.builditem.ModuleDirBuildItem;
+import io.quarkus.deployment.pkg.builditem.NativeImageBuildItem;
+import io.quarkus.deployment.util.ExecUtil;
+
+public class DockerBuildStep {
+
+    private static final Logger log = Logger.getLogger(DockerBuildStep.class);
+
+    @BuildStep(onlyIf = DockerBuild.class, onlyIfNot = NativeBuild.class)
+    public DockerImageResultBuildItem dockerBuildFromJar(ApplicationInfoBuildItem app, ModuleDirBuildItem moduledir,
+            Optional<DockerImageBuildItem> dockerImage,
+            JarBuildItem artifact) {
+        log.info("Building docker image for jar.");
+        ExecUtil.exec(moduledir.getPath().toFile(),
+                "docker", "build",
```

Comment:
We need to make this configurable, so we are not tied to docker. We also should provide the option of using some default docker files if there is none in the project.